### ISSUE TITLE
Adding an item under /sitecore/templates

### ISSUE THREAD

**Issue by kamsar** [sergeyshushlyapin/Sitecore.FakeDb#99]
I've got a test where I'd like to fake out a template under `/sitecore/templates/system`.

Based on [this wiki article](https://github.com/sergeyshushlyapin/Sitecore.FakeDb/wiki/creating-an-item-under-system) I used the following to attempt to tack the system item on under the default `/sitecore/templates` that comes with FakeDb, so that the code I have that creates the template could run.

    db.Add(new DbItem("System")
	{
		ParentID = ItemIDs.TemplateRoot
	});

    var item = db.GetItem("/sitecore/templates/system");

However executing this code seems to cause an exception resolving paths when I get the item?

```
System.InvalidOperationExceptionSequence contains more than one matching element
   at System.Linq.Enumerable.SingleOrDefault(IEnumerable`1 source, Func`2 predicate)
   at Sitecore.FakeDb.Data.Engines.DataCommands.ResolvePathCommand.DoExecute()
   at Sitecore.Data.Engines.EngineCommand`2.Execute()
   at Sitecore.Data.Managers.ItemProvider.GetItem(String itemPath, Language language, Version version, Database database, SecurityCheck securityCheck)
   at Sitecore.Data.Managers.ItemManager.GetItem(String itemPath, Language language, Version version, Database database)
   at x.Tests.Areas.xTests.CreateTestStore(Db db) in xTests.cs
```

**Comment by kamsar** [comment]
This is with 0.31.3 and Sitecore 8.1

**Comment by sergeyshushlyapin** [comment]
Nice catch, thanks for reporting!

The root of the problem is in the following. There are two items with the same name; the 'get item' call fails trying to return a *single* one. There is the more general scenario to reproduce:

```csharp
    [Fact]
    public void ShouldGetFirstItemByPathIfThereAreTwoSiblingsWithTheSameName()
    {
      using (var db = new Db
        {
          new DbItem("one-of-two"),
          new DbItem("one-of-two")
        })
      {
        db.GetItem("/sitecore/content/one-of-two");
      }
    }
```

**Comment by sergeyshushlyapin** [comment]
Fixed in [v0.31.5](https://github.com/sergeyshushlyapin/Sitecore.FakeDb/releases/tag/v0.31.5).
